### ISSUE TITLE
`mount -t tmpfs - /tmp` -> "found invalid line"

### ISSUE THREAD

**Issue by mralusw** [muesli/duf#153]
... presumably because the parser splits on *all* dashes and then checks for exactly two fields. It should. instead, split on `-` with maximum two fields.

Yes, I like to call my tmpfs mounts `-`.

**Comment by IGLOU-EU** [comment]
I can reproduce the bug

```
$ df -h
-                                 16G     0   16G   0% /mnt/a
a-b                               16G     0   16G   0% /mnt/b
```

```
$ duf
found invalid mountinfo line in file /proc/self/mountinfo: 557 27 0:73 / /mnt/a rw,relatime shared:284 - tmpfs - rw,inode64
```

**Comment by mralusw** [comment]
Still present in 0.8.0 I'm afraid.

Why doesn't `duf` skip mountinfo lines it doesn't grok? Why exit if a line is unparsable?

I even tried `duf -only-fs /`, and it still won't work.

**Comment by IGLOU-EU** [comment]
@mralusw Filtering stage are like a post-processing, so duf fail before this stage.
It may be more efficient to apply filtering before the processing. But this can be complicated, because every kernel (darwin, bsd, linux ...) has a dedicated mountinfo parser.

I'm still rewriting the mountinfo parsing, I plan to finish that this week.
If it's an urgency, you can try build duf from this commit e35e7029203758562df4f0f12902883c49b0a582

**Comment by mralusw** [comment]
No urgency, I was just curious. For example in https://github.com/muesli/duf/blob/0d660c36500629a0bbc1cc56695ad1821d23993c/mounts_linux.go#L38
why make that a global failure, instead of adding it to `warnings`?

**Comment by muesli** [comment]
Fixed by #154, in master now.
